src/ComposableController.ts

```diff
@@ -1,16 +1,34 @@
 import BaseController from './BaseController';
+import {
+  RestrictedControllerMessenger,
+  EventConstraint,
+} from './ControllerMessenger';
 
 /**
  * List of child controller instances
  */
-export type ControllerList = BaseController<any, any>[];
+export type ControllerList = (
+  | BaseController<any, any>
+  | { name: string; state: Record<string, unknown> }
```

Comment:
A minimal BaseControllerV2-like shape was used here, because the validation used on the BaseControllerV2 generic parameters prevented using that directly without knowing the exact state shape.

Reply:
This is probably worth documenting in the docstring of the type.

Reply:
Huh, good catch. This has reminded me to look at doc coverage, which is sadly not great throughout this entire PR! I will fix this.

Reply:
Documented in 089ae86106bfc7507e2e1412b616f13fe2cc520c